python/lsst/qa/explorer/dataset.py

```diff
@@ -1,806 +1,329 @@
+        id1, id2 = self.match_inds
 
-        label : str
-            Desired label of points.
+        vdims = self.vdims
+        v1 = df1.loc[id1, vdims]
+        v2 = df2.loc[id2, vdims]
+        v2.index = v1.index
 
-        filter_range, flags, bad_flags : dict, list, list
-            Parameters controlled by the `filter_stream` parameter
-            of `color_fit_explore`.
+        df = df1.copy()
+        df[vdims] = self._combine_operation(v1, v2)
+        df['match_distance'] = self.match_distance
 
-        x_range, y_range : 
-            Arguments required for datashaded map to be dynamic when passed to a DynamicMap        
-            (though right now this particular element does not get datashaded)
+        self._df = df
 
-        bounds : (l,b,r,t)
-            Bounds of selection box.
+    @property
+    def flags(self):
+        return self.data1.flags
 
-        order : int
-            Order of polynomial fit
-        """
+    def _getDims(self):
+        kdims, vdims = self.data1._getDims()
 
-        if mag is None:
-            mag = self.mag_names[0]
-
-        c1 = '{}_{}'.format(*colors[0:2])
-        c2 = '{}_{}'.format(*colors[1:3])
-
-        dset = self.get_color_ds(mag).select(x=(0,xmax), label=label)
-        if filter_range is not None:
-            dset = filter_dset(dset, filter_range=filter_range, flags=flags, bad_flags=bad_flags)
-
-        pts = dset.to(hv.Points, kdims=[c1, c2], groupby=[]).redim.range(**{c1:(-0.2,1.5),
-                                                                            c2:(-0.2,1.5)})
-        # Fit selected region to polynomial and plot
-        if bounds is None:
-            fit = hv.Curve([])
-        else:
-            l,b,r,t = bounds
-            subdf = pts.data.query('({0} < {4} < {2}) and ({1} < {5} < {3})'.format(l,b,r,t,c1,c2))
-            coeffs = np.polyfit(subdf[c1], subdf[c2], order)
-            x_grid = np.linspace(subdf[c1].min(), subdf[c1].max(), 100)
-            y_grid = np.polyval(coeffs, x_grid)
-    #         print(x_grid, y_grid)
-            fit = hv.Curve(np.array([x_grid, y_grid]).T).opts(style={'color':'black', 'width':3})
-            print(fit)
-        return pts * fit
-    #     return dynspread(datashade(pts, dynamic=False, x_range=x_range, y_range=y_range)) * fit
-
-    def color_fit_explore(self, xmax_range=np.arange(18,26.1,0.5), filter_stream=None):
-        """Dynamic map exploring polynomial fit to selected range of color-color plot
-
-        Parameters
-        ----------
-        xmax_range : array
-            Array of max magnitude values for slider widget
-
-        filter_stream : explorer.plots.FilterStream
-            Stream of constraints that controls what data to display.  Useful to link
-            multiple plots together
+        # Replace the *magDiff vdims with *mag
+        magDiffDims = [dim for dim in vdims if re.search('(.+_mag)Diff$', dim)]
+        magDims = [dim[:-4] for dim in magDiffDims]
```

Comment:
It's the `"cModel_mag"` part of `"cmodel_magDiff"`, e.g.  This is meant to deal with the rather specific situation of matching two `QADatasets` that have `*_magDiff` columns (e.g. `cmodel - psf` mags).  Rather than looking at the difference of the difference (which is a potentially confusing quantity), this changes to look at the difference between the datasets.   Again, it's a very specific situation and requires the `_magDiff` naming convention, but I did it because these `magDiff`-type columns are relatively common.  Probably the column naming convention for these magdiff functors should be documented somewhere.